test/cocotb/osdtestlib/debug_interconnect.py

```diff
@@ -165,78 +244,75 @@ def read_register(self, dut, destination, source, reg_width, address,
+            subtype = PACKET_SUBTYPE_REQ_READ_REG_16
+            expected_response = PACKET_SUBTYPE_RESP_READ_REG_SUCCESS_16
+        elif word_width == 32:
+            subtype = PACKET_SUBTYPE_REQ_READ_REG_32
+            expected_response = PACKET_SUBTYPE_RESP_READ_REG_SUCCESS_32
+        elif word_width == 64:
+            subtype = PACKET_SUBTYPE_REQ_READ_REG_64
+            expected_response = PACKET_SUBTYPE_RESP_READ_REG_SUCCESS_64
+        elif word_width == 128:
+            subtype = PACKET_SUBTYPE_REQ_READ_REG_128
+            expected_response = PACKET_SUBTYPE_RESP_READ_REG_SUCCESS_128
         else:
-            # display an error if an unsupported register width was chosen
-            raise TestFailure("Invalid register width parameter")
 
-        packet = DiPacket(destination = destination, source = source,
-                          packet_type = self.PACKET_TYPE_REG,
-                          packet_subtype = subtype, payload = address)
+            # display an error if an unsupported register width was chosen
+            raise RegAccessFailedException(message="An invalid register width\
+            parameter was chosen!")
 
+        tx_packet = DiPacket(destination=destination, source=source,
+                             packet_type=PACKET_TYPE_REG,
+                             packet_subtype=subtype, payload=address)
 
         # transmit the read request
-        yield self.send_packet(dut = dut, DiPacket = packet)
+        yield self.send_packet(dut=dut, packet=tx_packet)
+
+        response = []
+
+        # read the response
+        yield self.receive_packet(dut=dut, response=response)
 
-        # read in the response
-        yield self.receive_packet(dut = dut, response = response)
-        
         # display the response
         dut._log.info("Answer of the dut:")
-        dut._log.info(str(response))
+        dut._log.info('[{}]'.format(', '.join(hex(x) for x in response)))
 
-        # if the module responded with a successful read request, 
+        # fill the receive packet
+        rx_packet.from_raw2(flits=response)
+
+        # if the module responded with a successful read request,
         # let the user know
-        if ((response[2] & 0x3C00) >> 10) == expected_response:
+        if rx_packet.packet_subtype == expected_response:
             dut._log.info("Read process was successful!")
 
         # if it returned an error, output that error
-        elif ((response[2] & 0x3C00) >> 10) == \
-            self.PACKET_SUBTYPE_RESP_READ_REG_ERROR:
-            dut._log.info("An error occurred during the read process!")
+        elif rx_packet.packet_subtype == PACKET_SUBTYPE_RESP_READ_REG_ERROR:
+            raise RegAccessFailedException(message="An error occurred during\
+            the read process!")
 
         else:
-            dut._log.info("An unidentified error occurred!")
-
+            raise RegAccessFailedException(message="An unidentified error\
+            occurred!")
```

Comment:
Actually, that error is not "unidentified" -- you got a subtype which isn't what you expected.